views/domain_mode_eap_operations.py

```diff
@@ -0,0 +1,98 @@
+from common.ui_utils import ui_utils
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.common.by import By
+from time import sleep
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+'''
+
+Created on December 15, 2016
+
+@author: pyadav
+
+'''
+
+class domain_mode_eap_operations():
+
+    web_session = None
+
+    def __init__(self, web_session):
+
+        self.web_session = web_session
+        self.web_driver = web_session.web_driver
+
+    def find_domain_mode_eap_power(self):
+        self.web_session.web_driver.get("{}/middleware_domain/show_list".format(self.web_session.MIQ_URL))
+        assert ui_utils(self.web_session).waitForTextOnPage("Middleware Domains", 15)
+        self.web_session.web_driver.find_element_by_xpath("//td[contains(.,'master')]").click()
+        assert ui_utils(self.web_session).waitForTextOnPage("master (Summary)", 15)
+        self.web_session.web_driver.find_element_by_xpath("//td[contains(.,'Middleware Server Groups')]").click()
+        assert ui_utils(self.web_session).waitForTextOnPage("Server Group Name", 15)
+        self.web_session.web_driver.find_element_by_xpath("//td[contains(.,'main-server-group')]").click()
+        assert ui_utils(self.web_session).waitForTextOnPage("main-server-group (Summary)", 15)
+        self.web_session.web_driver.find_element_by_xpath("//td[contains(.,'Middleware Servers')]").click()
+        assert ui_utils(self.web_session).waitForTextOnPage("Server Name", 15)
+        self.web_session.web_driver.find_element_by_xpath("//td[contains(.,'server-one')]").click()
+        assert ui_utils(self.web_session).waitForTextOnPage("Properties", 15)
+        self.web_session.web_driver.find_element_by_xpath("//button[@title='Power']").click()
+
```

Comment:
Could domains.nav_to_all_middleware_server_groups be used to navigate to Server Groups?

Seems that each caller should find a server which is in a specific state.
For example, suspending an EAP that is already in suspend state would not be a valid test since the EAP is already in the state for which the poser operation is requesting.

If there are no Domain EAPs in desire state, the test case should gracefully handle such. We may want to start using features such as pytest.skip() to indicate that a test case was not run, in this case due to a Domain server not being available.